csrc/preseg_passes/allocation_order_inference.cpp

```diff
@@ -309,13 +316,16 @@ void inferenceAllocationOrder(
       // found multiple candidate with the same iterdomain count
       if (non_trivial_iter_count[tv] == non_bc_high_water_mark &&
           ref != nullptr) {
+        std::vector<IterDomain*> ref_alloc_non_trivial =
+            nonTrivialIterDomains(ref);
+        std::vector<IterDomain*> tv_alloc_non_trivial =
+            nonTrivialIterDomains(tv);
         // ensure that there's no ambiguity on permutation mapping from multiple
         // references. we need both ref candidates to have the same mapping on
         // allocation domain
-        for (auto i : c10::irange(ref->nDims())) {
+        for (auto i : c10::irange(ref_alloc_non_trivial.size())) {
```

Comment:
If you NVF_ERROR same size as suggested above, consider `zip` to simplify the code